### ISSUE TITLE
rustdoc renders renamed imports using the new name

### ISSUE THREAD

**Issue by camelid** [rust-lang/rust#81141]
rustdoc renders renamed imports (`use foo as bar;`) using the new, private name
(e.g., `bar` in `use foo as bar;`). This is confusing behavior since the new
name is an implementation detail. I think it should be using the original
item's name instead.

As an example, `chrono::Date::signed_duration_since` returns a `Duration`, but
the module it's defined in renames `Duration` to `OldDuration`. I would expect
rustdoc to still show the item's definition name, but instead it shows the
private `OldDuration` name.

**Comment by jyn514** [comment]
I don't think this should be changed until https://github.com/rust-lang/rust/issues/42066 is fixed. Otherwise you could have two items shown with the same name and namespace in the same scope.

**Comment by GuillaumeGomez** [comment]
Code to reproduce:

```rust
use core::time::Duration as Alias;

pub fn bar() -> Alias {
    Alias::new(0, 0)
}
```

I don't think we need to wait for https://github.com/rust-lang/rust/issues/42066 to be fixed as the fix for this doesn't impact it: if the re-export is not publicly reexported, we simply show the item underneath. When we will fix #42066, it will handle it the same.
